### PR TITLE
Cycle Sort added

### PR DESCRIPTION
**Pull request by SupriyaKumari08**
## Related Issue
- Cycle Sort - (#1359 )

## Proposed Changes
- Added code for cycle sort.
- Updated README.md file under Sorting Algorithms for Cycle Sort with algorithm, time complexity, practice problems etc.

## Additional Info
-The code & algorithm follows a simplistic approach hence beneficial to the maximum percentage of users.
-This code solves an open issue (https://github.com/Lakhankumawat/LearnCPP/issues/1359)
-I have tested these changes on VS code & online compiler onlinegdb.com using multiple inputs and matched the output with the expected output.
-This change will not affect or create conflict with any other code.
-Time complexity: O(N*N)
-Space complexity: O(1)

## Checklist
- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- I have added the code, algorithm, properties(of cycle sort), advantages, disadvantages & some practice problems for Cycle Sort.
- The code is added under Sorting Algorithms directory named 'CycleSort.cpp'
- The other information mentioned above regarding cycle sort is added in the README.md file under Sorting Algorithms directory.

- This algorithm aims to sort any given no. of elements in an array. 

- This change provide a useful alternative for sorting data.

- This issue solves (https://github.com/Lakhankumawat/LearnCPP/issues/1359)

- I have coded this problem in online  C++ compiler onlinegdb.com and tested the given problem on the various test cases including those with duplicate elements. The code passed all of them.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
| With duplicate elements  | Without duplicate elements    |
|cyclesort snippet1|
 |cyclesort snippet2|

### CODE REVIEW

File: S-SortingAlgorithms/README.md [hunk 1]
```diff
@@ -506,3 +513,115 @@ DNF Sort (Dutch National Flag Sorting) , This is the sorting method which is spe
...
+    }
+    cycleSort(arr,n);
+    cout<<"Sorted array:"<< endl;
+    for(int i=0; i < n; i++){
+          cout << arr[i] << " " ;
+    }
+    return 0;  
+}  
+    
+```
```
Comment (Lakhankumawat) [inline S-SortingAlgorithms/README.md:599]: remove this section and add psuedocode instead .

File: S-SortingAlgorithms/README.md [hunk 2]
```diff
@@ -506,3 +513,115 @@ DNF Sort (Dutch National Flag Sorting) , This is the sorting method which is spe
 ### Disadvantage
 
 - Problem was restricted by allowing the inspection of the color of each element only once.
+
+-----------
+
+### Cycle Sort
+-Cycle sort is a comparison based sorting algorithm which forces array to be factored into the number of cycles where each of them can be rotated to produce a sorted array.
```
Comment (Lakhankumawat) [inline S-SortingAlgorithms/README.md:520]: Add a working image inbetween explanation for sort 
You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

File: S-SortingAlgorithms/README.md (hunk 2 above)
Comment (SupriyaKumari08) [reply S-SortingAlgorithms/README.md:520]: On it👍

**Comment by SupriyaKumari08** [comment]
@Lakhankumawat I have modified the README file...Kindly let me know if everything is correct/accurate.

**Review by Lakhankumawat — APPROVED** [review]
## LGTM✨🎉

Good Contribution @SupriyaKumari08 
Keep Learning 👩‍💻
Keep Growing 🌱
Keep Shining ✨

**Comment by Lakhankumawat** [comment]
Your contribution is good , giving level 2 , congrats

**Comment by Lakhankumawat** [comment]
@all-contributors please add @SupriyaKumari08 code ,

**Pull request merged**